### ISSUE TITLE
`followComponent` appears incorrect on RC 12

### ISSUE THREAD

**Issue by anleac** [flame-engine/flame#854]
On the latest RC, `1.0.0-releasecandidate.12`, the cameras `followComponent` seems to be a bit broken on sudden movement change (especially after sudden movement change, said component also remains static, as per the repro case below).

Zoom is 1, and nothing too interesting with the camera except for world bounds which only allow vertical movement.

### The pretext
I have a simple app which tracks a player component moving vertically up, with the world bounds being the world bottom, and the sides of the phone (basically doodle jump). On player death, the player will respawn at the world bottom and I expect the camera to snap to this.

### The bug
On the latest version, it appears on the player reset to the bottom of the map (this also means the player is waiting at the bottom again), that the camera does not correct snap back to the bottom.

Note: Shifting back to `1.0.0-releasecandidate.11` fixes this issue.

I have set up a fairly simple app, you can see the entire code here. You can repro this, with the `1.0.0-releasecandidate.12` version. 

This repro app features a simple textComponent that's being followed (you can see the debug stats for it's position) and it's moving up by a pixel every update tick. On pressing tap, it'll reset the position and stop moving (and the camera is called to re-follow, though I'm not even sure this should be needed). You can notice that the camera does not correctly re-center on RC12.

```
import 'dart:math';

import 'package:flame/components.dart';
import 'package:flame/game.dart';
import 'package:flame/gestures.dart';
import 'package:flame/palette.dart';
import 'package:flutter/material.dart';

void main() {
  runApp(
    GameWidget(game: MyGame()),
  );
}

class MyGame extends BaseGame with TapDetector {
  late final double _upperWorldCameraBound = -pow(2, 60).toDouble();

  Rect get defaultWorldBounds => Rect.fromLTRB(0, _upperWorldCameraBound, size.x, size.y);
  Vector2 get defaultTextPosition => Vector2(size.x / 2, size.y / 2);

  late final TextComponent _centerText = TextComponent(
    "Follow me",
    textRenderer: TextPaint(config: TextPaintConfig(fontSize: 28.0, fontFamily: "Baloo", color: BasicPalette.red.color)),
    position: defaultTextPosition)
    ..anchor=Anchor.center;

  late final TextComponent _positionText = TextComponent(
    '',
    textRenderer: TextPaint(config: TextPaintConfig(fontSize: 16.0, fontFamily: "Baloo", color: BasicPalette.red.color)),
    position: Vector2.all(30))
    ..anchor=Anchor.topLeft
    ..isHud=true;

  late final TextComponent _infoText = TextComponent(
    'Tap to stop moving and re-center',
    textRenderer: TextPaint(config: TextPaintConfig(fontSize: 16.0, fontFamily: "Baloo", color: BasicPalette.red.color)),
    position: Vector2(30, 60))
    ..anchor=Anchor.topLeft
    ..isHud=true;

  bool _movingUp = true;

  @override
  Future<void> onLoad() async {
    add(_centerText);
    add(_positionText);
    add(_infoText);

    camera.followComponent(_centerText, worldBounds: defaultWorldBounds);
  }

  @override
  void update(double dt) {
    super.update(dt);

    if (_movingUp) {
      _centerText.position.y -= 1;
    }

    _positionText.text = "Text pos: " + _centerText.position.toString();
  }

  @override
  void onTap() {
    _movingUp = false;
    _centerText.position = defaultTextPosition;
    camera.followComponent(_centerText, worldBounds: defaultWorldBounds);
    super.onTap();
  }
}
```

**Comment by spydon** [comment]
Thanks for the reproduction example @anleac!
I have fixed the issue in #857, could you verify that it works for you too by depending on that branch?

Simply put this in your `pubspec.yaml` instead of the flame version:
```yaml
dependencies:
  flame:
    git:
      url: git@github.com:flame-engine/flame.git
      path: packages/flame
      ref: spydon.fix-transform-matrix-cache
```

**Comment by anleac** [comment]
Thanks @spydon - I can confirm this fix works 🤝
